src/transformers/modeling_gguf_pytorch_utils.py

```diff
@@ -307,23 +355,11 @@ def load_gguf_checkpoint(gguf_checkpoint_path, return_tensors=False):
         ffn_norm_name = "ffn_norm"
         qkv_bias = any(bias_name in tensor.name for tensor in reader.tensors for bias_name in attn_bias_name)
         use_parallel_residual = any(ffn_norm_name in tensor.name for tensor in reader.tensors)
-        parsed_parameters["config"]["qkv_bias"] = qkv_bias
+        parsed_parameters["config"]["use_qkv_bias"] = qkv_bias
         parsed_parameters["config"]["use_parallel_residual"] = not use_parallel_residual
 
-    model_size = ""
-    # extract the number of params from file name as architectures can differ ;
-    # eg. for falcon : `...falcon-7b-...`
-    if "falcon" in architecture:
-        gguf_file_name = gguf_checkpoint_path.split("/")[-1].lower()
-        m = re.search(r"-\d+b-", gguf_file_name)  # regex to catch `-7b-`
-        if m is None:
-            raise ValueError(
-                f"From file name, cannot determine the number of parameters for {architecture} architecture"
-            )
-        model_size = m.group().strip("-")  # only keeps `7b`
-
-    if architecture + model_size not in GGUF_SUPPORTED_ARCHITECTURES:
-        raise ValueError(f"Architecture {architecture + model_size} not supported")
+    if architecture not in GGUF_SUPPORTED_ARCHITECTURES:
+        raise ValueError(f"Architecture {architecture} not supported")
```

Comment:
Maybe ask him to download gguf from source to see if it is supported ?

Reply:
I think we should ask for gguf installation from source in `get_gguf_hf_weights_map` instead of here:
```python
    for key, value in MODEL_ARCH_NAMES.items():
        if value == model_type:
            arch = key
            break
    if arch is None:
        raise RuntimeError(f"Unknown gguf model_type: {model_type}")
```

Because `GGUF_SUPPORTED_ARCHITECTURES` here is still based on our `GGUF_CONFIG_MAPPING`, which still needs to be added manually when adding new GGUF model support.